pkg/controller/node_availability.go

```diff
@@ -57,15 +56,9 @@ func (r *ReconcileNodeAvailability) Reconcile(ctx context.Context, request recon
 	if err == nil {
 		if !n.Spec.Unschedulable {
 			//Node is schedulable, proceed to attempt creation of replica attachment
-			if atomic.SwapInt32(&r.controllerSharedState.processingReplicaRequestQueue, 1) == 0 {
-				logger.Info("Node is now available. Will requeue failed replica creation requests.")
-				err := r.controllerSharedState.tryCreateFailedReplicas(ctx, nodeavailability)
-				atomic.StoreInt32(&r.controllerSharedState.processingReplicaRequestQueue, 0)
-				if err != nil {
-					return reconcile.Result{Requeue: false}, nil
-				}
-				return reconcile.Result{Requeue: false}, nil
-			}
+			logger.Info("Node is now available. Will requeue failed replica creation requests.")
+			err := r.controllerSharedState.tryCreateFailedReplicas(ctx, nodeavailability)
```

Comment:
Since replica creation operation will be pushed to a queue for later processing, I'm not sure it makes sense to use the reconciler context here. A context with a long timeout derived from `context.Background()` seems like a more appropriate choice. This is an issue with the existing code, so I am fine deferring this to another PR (and more discussion).